### PR TITLE
Add a table-based view for the performance data

### PR DESCRIPTION
**Pull request by gregtatum**
This includes the unit and whether lower or higher is better. I felt a
bit confused looking at the results, and not knowing what the individual
values meant. This change also presents everything in a table format
which should make it easier to scan and understand.

I ran these changes by some users of this view, and handled some of
their early feedback. This included removing a graph that wasn't very
helpful. There might be some opportunities in the future to add more
details to the columns to help in making navigating the data easier.
However, this is not a replacement for the comparison view, but instead
the motivation is for making the individual test runs easier to
understand.

Useful link for testing:
 * [Raptor](https://treeherder.mozilla.org/jobs?repo=autoland&revision=43e0f933ad51ba77e5528a7125a958cee7bf4612&searchStr=raptor&selectedTaskRun=bXfMyEIiSR69F0iI7tzILg.0)
 * [Talos](https://treeherder.mozilla.org/jobs?repo=autoland&revision=43e0f933ad51ba77e5528a7125a958cee7bf4612&searchStr=talos&selectedTaskRun=bXfMyEIiSR69F0iI7tzILg.0)
 * [Browsertime](https://treeherder.mozilla.org/jobs?repo=autoland&revision=43e0f933ad51ba77e5528a7125a958cee7bf4612&searchStr=browsertime&selectedTaskRun=bXfMyEIiSR69F0iI7tzILg.0)

## Screenshots

image

image

image

## Before view

image

### CODE REVIEW

**Comment by camd** [comment]
@gregtatum : Yes, I can take a look at this in a bit.  I was swamped with a few other things last week.

File: ui/css/treeherder-details-panel.css
Comment (camd) [inline ui/css/treeherder-details-panel.css:426]: Would you mind trying to get this layout using the `bootstrap` `utilities` like `pl-1`,`pr-1` for padding and `text-nowrap` classes instead of custom CSS?  I realize some custom CSS is necessary, but we are trying to minimize it.

We also have lots of helper utilities classes in `treehrder-custom-styles.css` like `small-text` is `font-size: 12`, etc.

File: ui/css/treeherder-details-panel.css
Comment (camd) [inline ui/css/treeherder-details-panel.css:448]: Making this a `reactstrap` `Row` component might handle the horizontal display.

**Review by camd — CHANGES_REQUESTED** [review]
This looks really nice!  I don't tend to use this panel myself, so I'll leave it to ionut or the other perf stakeholders to decide if they like the design.  But I'd be very surprised if everybody didn't think this was a vast improvement.  :)

The only issue, as mentioned, is perhaps using more of the bootstrap utilities classes.

File: ui/css/treeherder-details-panel.css
Comment (gregtatum) [reply ui/css/treeherder-details-panel.css:448]: I wrapped the `<a href>` in a row, and it doesn't really solve all the layout issues here compared to the custom CSS. I'd prefer to be precise in how this works around the bootstrap rules, as it's not clear to me that the rules in `Row` make it any simpler.

image

Comment (camd) [reply ui/css/treeherder-details-panel.css:448]: Oh, I should have tried this.  Sorry.  I was way off on the `<Row>` element...

That being said, I tried this with the `<a>` element class having: `btn btn-outline-darker-secondary btn-sm performance-panel-view-button py-0` and then the `performance-panel-view-button` just setting the line-height to 17.  It looks like this for me:

Screen Shot 2020-11-12 at 4 29 03 PM

**Review by ionutgoldan — CHANGES_REQUESTED** [review]
I'm very happy with the new Javascript functionality. Totally a fan of that.

However, please fix the CSS linting reported by the yarn tests. Once this gets addressed, you have my r+!

File: ui/css/treeherder-details-panel.css
Comment (camd) [reply ui/css/treeherder-details-panel.css:434]: > My process so far has been to design the space how I think it should look, and then I back off the custom style rules to hunt for bootstrap rules that make sense. Then in reviews, I'm getting lots of feedback on how to use even more bootstrap styles, but it degrades the initial designed layout. This doesn't really feel like an efficient process, and I'm not seeing how the bootstrap classes really help here.

It is certainly not my intention to make things harder for you to volunteer your time with us.  :). I apologize for the frustration there.  

The reasoning and motivation, such as they are, is that we ended up with a ton of complex and overlapping `CSS` in Treeherder that grew over the years and we're trying to reduce that now.  It ended up conflicting with itself at times and painted us into some corners.  So we're just trying to reuse what's already there in Bootstrap when possible and if it does the job right.  But you're already doing that here, and it totally makes sense that using custom CSS to make sure the job is done right is the way to go.  So thanks for bearing with my questions and suggestions.  :)

**Review by camd — APPROVED** [review]
Thanks for doing this.  It is a huge improvement!  :)

**Comment by camd** [comment]
It looks like @ionutgoldan has given his approval when the lining issues were fixed, which they are.  So I'll go ahead and merge.

File: ui/css/treeherder-details-panel.css
Comment (gregtatum) [reply ui/css/treeherder-details-panel.css:434]: Thanks for the explanation and context here. I re-read my comments, and they came across as a bit aggressive. Sorry about that. Thanks for your patience and help getting this merged!

Comment (camd) [reply ui/css/treeherder-details-panel.css:434]: Thanks and no worries!  All is good.  :)

**Pull request merged**
